src/drivers/amd/rembrandt/acp_sw_audio_dma.c

```diff
@@ -0,0 +1,640 @@
+	sw_dev[SDW0_ACP_SW_AUDIO_TX_EN_CH].sw_dev_ringbuff_size = ACP_AUDIO_TX_RINGBUFSIZE;
+	sw_dev[SDW0_ACP_SW_AUDIO_TX_EN_CH].sw_dev_dma_size = ACP_AUDIO_TX_DMA_SIZE;
+	sw_dev[SDW0_ACP_SW_AUDIO_TX_EN_CH].sw_dev_dma_watermark = ACP_AUDIO_TX_INTR_WATERMARK_SIZE;
+	sw_dev[SDW0_ACP_SW_AUDIO_TX_EN_CH].sw_dev_dma_intr_status = ACP_DSP0_INTR_STAT;
+	sw_dev[SDW0_ACP_SW_AUDIO_TX_EN_CH].sw_dev_dma_intr_cntl = ACP_DSP0_INTR_CNTL;
+	sw_dev[SDW0_ACP_SW_AUDIO_TX_EN_CH].statusindex = 5;
+
+	sw_dev[SDW0_ACP_SW_BT_RX_EN_CH].sw_dev_en = ACP_SW_BT_RX_EN;
+	sw_dev[SDW0_ACP_SW_BT_RX_EN_CH].sw_dev_en_status = ACP_SW_BT_RX_EN_STATUS;
+	sw_dev[SDW0_ACP_SW_BT_RX_EN_CH].sw_dev_fifo_addr = ACP_BT_RX_FIFOADDR;
+	sw_dev[SDW0_ACP_SW_BT_RX_EN_CH].fifo_addr = SW0_BT_RX_FIFO_ADDR;
+	sw_dev[SDW0_ACP_SW_BT_RX_EN_CH].sw_dev_fifo_size = ACP_BT_RX_FIFOSIZE;
+	sw_dev[SDW0_ACP_SW_BT_RX_EN_CH].fifo_size = SW0_BT_FIFO_SIZE;
+	sw_dev[SDW0_ACP_SW_BT_RX_EN_CH].sw_dev_ringbuff_addr = ACP_BT_RX_RINGBUFADDR;
+	sw_dev[SDW0_ACP_SW_BT_RX_EN_CH].sw_dev_ringbuff_size = ACP_BT_RX_RINGBUFSIZE;
+	sw_dev[SDW0_ACP_SW_BT_RX_EN_CH].sw_dev_dma_size = ACP_BT_RX_DMA_SIZE;
+	sw_dev[SDW0_ACP_SW_BT_RX_EN_CH].sw_dev_dma_watermark = ACP_BT_RX_INTR_WATERMARK_SIZE;
+	sw_dev[SDW0_ACP_SW_BT_RX_EN_CH].sw_dev_dma_intr_status = ACP_DSP0_INTR_STAT;
+	sw_dev[SDW0_ACP_SW_BT_RX_EN_CH].sw_dev_dma_intr_cntl = ACP_DSP0_INTR_CNTL;
+	sw_dev[SDW0_ACP_SW_BT_RX_EN_CH].statusindex = 2;
+
+	sw_dev[SDW0_ACP_SW_BT_TX_EN_CH].sw_dev_en = ACP_SW_BT_TX_EN;
+	sw_dev[SDW0_ACP_SW_BT_TX_EN_CH].sw_dev_en_status = ACP_SW_BT_TX_EN_STATUS;
+	sw_dev[SDW0_ACP_SW_BT_TX_EN_CH].sw_dev_fifo_addr = ACP_BT_TX_FIFOADDR;
+	sw_dev[SDW0_ACP_SW_BT_TX_EN_CH].fifo_addr = SW0_BT_TX_FIFO_ADDR;
+	sw_dev[SDW0_ACP_SW_BT_TX_EN_CH].sw_dev_fifo_size = ACP_BT_TX_FIFOSIZE;
+	sw_dev[SDW0_ACP_SW_BT_TX_EN_CH].fifo_size = SW0_BT_FIFO_SIZE;
+	sw_dev[SDW0_ACP_SW_BT_TX_EN_CH].sw_dev_ringbuff_addr = ACP_BT_TX_RINGBUFADDR;
+	sw_dev[SDW0_ACP_SW_BT_TX_EN_CH].sw_dev_ringbuff_size = ACP_BT_TX_RINGBUFSIZE;
+	sw_dev[SDW0_ACP_SW_BT_TX_EN_CH].sw_dev_dma_size = ACP_BT_TX_DMA_SIZE;
+	sw_dev[SDW0_ACP_SW_BT_TX_EN_CH].sw_dev_dma_watermark = ACP_BT_TX_INTR_WATERMARK_SIZE;
+	sw_dev[SDW0_ACP_SW_BT_TX_EN_CH].sw_dev_dma_intr_status = ACP_DSP0_INTR_STAT;
+	sw_dev[SDW0_ACP_SW_BT_TX_EN_CH].sw_dev_dma_intr_cntl = ACP_DSP0_INTR_CNTL;
+	sw_dev[SDW0_ACP_SW_BT_TX_EN_CH].statusindex = 3;
+
+	atomic_init(&dma->num_channels_busy, 0);
+	return 0;
+}
+
+static int acp_dai_sw_audio_dma_remove(struct dma *dma)
+{
+	if (!dma->chan) {
+		tr_err(&acp_sw_audio_tr, "remove called without probe,it's a no-op");
+		return 0;
+	}
+
+	rfree(dma->chan);
+	dma->chan = NULL;
+	return 0;
+}
+
+static int acp_dai_sw_audio_dma_get_data_size(struct dma_chan_data *channel,
+					      uint32_t *avail, uint32_t *free)
+{
+	if (channel->direction == DMA_DIR_MEM_TO_DEV) {
+		*free = (sw_audio_buff_size_playback >> 1);
+		*avail = (sw_audio_buff_size_playback >> 1);
```

Comment:
no need for parentheses above and below

Reply:
maybe explain the math and what the division by two means?

Reply:
done.
half the playback buffer size is available.